app/platform/common.c

```diff
@@ -67,22 +67,54 @@ uint32_t platform_flash_get_num_sectors(void)
 #endif // #ifdef INTERNAL_FLASH_SECTOR_SIZE
 }
 
+static uint32_t allocated = 0;
+static uint32_t phys_flash_used_end = 0;  //Phyiscal address of last byte in last flash used sector 
+
+uint32_t platform_flash_reserve_section( uint32_t regsize, uint32_t *start )
```

Comment:
Don't take this the wrong way Terry, but I really don't think this is the way to go. We've kept on bolting on various flash allocation bits over the years. I think it's time we bite the bullet and switch to a partition table like on the ESP32. That way we can easily tell where the spiffs image(s?) are, the LFS area, SDK reserved areas, etc.

With the approach here we're effectively introducing race/ordering problems where things may move around unexpectedly in flash. Having a partition table would keep things a lot cleaner and safer.

Reply:
See comment above.  This is really an issue it it's own right and maybe worth tracking as such.